bftengine/include/bftengine/Replica.hpp

```diff
@@ -35,6 +35,11 @@ enum MsgFlag : uint8_t {
   HAS_PRE_PROCESSED_FLAG = 0x4
 };
 
+class IControlHandler {
```

Comment:
Why does this need to be an abstract class? 

I ask because it looks like the handlers are directly included (composition) via the functions.

Reply:
@yontyon and I discussed this, and it makes sense as an abstract class because right now we expect the handlers to be implemented all or nothing by application code.